node/resolvers/Mutations/Users.ts

```diff
@@ -183,26 +183,29 @@ const Users = {
       const result = await storefrontPermissionsClient.getRole(user.roleId)
       const roleSlug = result?.data?.getRole?.slug
 
-      let permitted = false
+      const canImpersonateUsersWithCostCenterPermission =
+        !roleSlug.includes('admin') &&
+        storefrontPermissions?.permissions?.includes(
+          'impersonate-users-costcenter'
+        ) &&
+        user?.costId ===
+          sessionData?.namespaces['storefront-permissions']?.costcenter?.value
 
-      if (!roleSlug.includes('sales')) {
```

Comment:
It looks like the code was not allowing `sales` role to impersonate on purpose. Do we have any idea why?

Reply:
I don't know why, but in the [documentation](https://developers.vtex.com/docs/apps/vtex.b2b-organizations) it is specified that the role has access. Do you know the reasons @wender or @arturmagalhaesjr?
![Screenshot 2023-10-06 at 10 37 34](https://github.com/vtex-apps/b2b-organizations-graphql/assets/5332361/5686e962-9c2c-47df-b389-e93fdee6c24e)